### PR TITLE
DEV-77: Enable mac launcher builds

### PR DESCRIPTION
**Pull request by jherico**
https://highfidelity.atlassian.net/browse/DEV-77

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19781/HighFidelity-Beta-PR15681-b6028c2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19781/HighFidelity-Beta-Interface-PR15681-b6028c2.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19781/HighFidelity-Beta-PR15681-b6028c2.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19781/HighFidelity-Quest-Beta-PR15681-b6028c2.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19786/HighFidelity-Beta-PR15681-05df59d.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19786/HighFidelity-Quest-Beta-PR15681-05df59d.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19786/HighFidelity-Beta-PR15681-05df59d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19786/HighFidelity-Beta-Interface-PR15681-05df59d.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19789/HighFidelity-Beta-PR15681-0876a78.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19789/HighFidelity-Quest-Beta-PR15681-0876a78.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19789/HighFidelity-Beta-PR15681-0876a78.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19789/HighFidelity-Beta-Interface-PR15681-0876a78.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19789/HighFidelity-Beta-PR15681-0876a78.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19789/HighFidelity-Beta-Interface-PR15681-0876a78.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19791/HighFidelity-Beta-PR15681-8e1f4b5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19791/HighFidelity-Beta-Interface-PR15681-8e1f4b5.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19791/HighFidelity-Beta-PR15681-8e1f4b5.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19791/HighFidelity-Quest-Beta-PR15681-8e1f4b5.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19791/HighFidelity-Beta-PR15681-8e1f4b5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19791/HighFidelity-Beta-Interface-PR15681-8e1f4b5.exe

File: launchers/darwin/CMakeLists.txt [hunk 1]
```diff
@@ -53,10 +53,21 @@ set_target_properties(${this_target} PROPERTIES
...
+    endif()
+endfunction()
+
 add_executable(${PROJECT_NAME} MACOSX_BUNDLE ${src_files})
 set_target_properties(${PROJECT_NAME} PROPERTIES OUTPUT_NAME ${APP_NAME})
-
-
+set_from_env(LAUNCHER_HMAC_SECRET LAUNCHER_HMAC_SECRET "")
+target_compile_definitions(${PROJECT_NAME} PRIVATE LAUNCHER_HMAC_SECRET="${LAUNCHER_HMAC_SECRET}")
+message("HMAC secret ${LAUNCHER_HMAC_SECRET}")
```
Comment (jherico) [inline launchers/darwin/CMakeLists.txt]:
```suggestion

```

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19799/HighFidelity-Beta-PR15681-a96d57e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19799/HighFidelity-Beta-Interface-PR15681-a96d57e.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19799/HighFidelity-Beta-PR15681-a96d57e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19799/HighFidelity-Quest-Beta-PR15681-a96d57e.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19799/HighFidelity-Beta-PR15681-a96d57e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19799/HighFidelity-Beta-Interface-PR15681-a96d57e.exe

**Pull request merged**